web/src/components/molecules/Common/Form/AssetItem/index.tsx

```diff
@@ -96,19 +98,37 @@ const AssetItem: React.FC<Props> = ({
   const [asset, setAsset] = useState<ItemAsset>();
   const selectedAssetRef = useRef<ItemAsset>();
 
+  const defaultValueGet = useCallback(
+    async (id: string) => {
+      const fileName = await onGetAsset?.(id);
+      if (fileName) setAsset({ id, fileName });
+    },
+    [onGetAsset],
+  );
+
   useEffect(() => {
-    const defaultAsset = itemAssets?.find(itemAsset => itemAsset.id === value);
-    if (defaultAsset) {
-      setAsset(defaultAsset);
-    } else {
+    if (loadingAssets) return;
+    const assetInfo = itemAssets?.find(itemAsset => itemAsset.id === value);
+    if (assetInfo) {
+      setAsset(assetInfo);
+    } else if (selectedAssetRef.current) {
       setAsset(selectedAssetRef.current);
+    } else if (value) {
+      defaultValueGet(value);
+    } else {
+      setAsset(undefined);
     }
-  }, [itemAssets, value]);
+  }, [defaultValueGet, value, loadingAssets, itemAssets]);
```

Comment:
Can you rethink this login?

Reply:
I think this is the only way to avoid throwing getAsset query in vain.
If the result of the getItem query could include a default asset name, it would be simple a bit.